Fix/elements display

MatMT opened the pull request:
---
## Summary

This pull request introduces several updates to the OrganiTask application, focusing on enhancing the Kanban board functionality, improving UI elements, and adding new features for better user experience and maintainability. 

### Kanban Board Enhancements:
* Updated the `KanbanColumnPanel` to use `ColorUtil` for parsing colors and adjusted the scroll behavior for better usability (`OrganiTask/Forms/Controls/KanbanColumnPanel.cs`). [[1]](diffhunk://#diff-7c10b5456fef81310ff8fb2081355f4ddee2f321f90dcb2be94dcc7acdd3c89bR30-R52) [[2]](diffhunk://#diff-7c10b5456fef81310ff8fb2081355f4ddee2f321f90dcb2be94dcc7acdd3c89bL56-R72) [[3]](diffhunk://#diff-7c10b5456fef81310ff8fb2081355f4ddee2f321f90dcb2be94dcc7acdd3c89bL69-L89)

### New Features:
* Added a `CategoryExists` method in `DashboardController` to verify the existence of a category in a dashboard (`OrganiTask/Controllers/DashboardController.cs`).
* Introduced a `Tags` property in `TaskViewModel` to associate tags with tasks (`OrganiTask/Entities/ViewModels/TaskViewModel.cs`).
* Now it's possible modify default category and tags when you are creating a new Dashboard.
* Allow only one instance per Dashboard. If the user tries to open it again and it is not closed, bring it to focus.

### UI Improvements:
* Adjusted the `TaskCardPanel` height to accommodate additional information (`OrganiTask/Forms/Controls/TaskCardPanel.cs`).
* Updated the `DashboardsManagement` form to refine default category options and improve text clarity (`OrganiTask/Forms/DashboardsManagement.Designer.cs`). [[1]](diffhunk://#diff-f584535b5cdca1e0db8eb35153b63383288e80e6b012a704b0b28513c2e59791L142-R147) [[2]](diffhunk://#diff-f584535b5cdca1e0db8eb35153b63383288e80e6b012a704b0b28513c2e59791L171-R169)

### Codebase Cleanup:
* Removed unused methods and redundant code from `KanbanColumnPanel`, simplifying its implementation (`OrganiTask/Forms/Controls/KanbanColumnPanel.cs`).
* Cleaned up unused imports and adjusted code structure for better readability (`OrganiTask/Forms/DashboardSettings.cs`).

---

## Demo Video 


https://github.com/user-attachments/assets/938fc3c4-8a5c-4e51-b736-4db3f63cf51d

MatMT committed:
Column size and horizontal scroll hidden

MatMT committed:
Show Kanban Dashboard always and maximisation of the window created

MatMT committed:
DashboardManager added to open and make focus on minimized or non active windows KanbanForms

MatMT committed:
fix: lblHeader display on large Dashboards names

MatMT committed:
WIP: fixed desplay cboSort when there are no categories in the dashboard and enabled default category edit

Peding: Create and Delete for default creation tags

MatMT committed:
Show conditional message when the user is going to delete the last category on the dashboard

MatMT committed:
feat: Now it's possible modify default category and tags

Double click and confirmation to delete one tag on TaglistBox

MatMT committed:
feat: ForeColor is now dynamic — it depends on the background color. The refreshDashboard method now verifies the category to render the data correctly.

Pending: Bug when renaming a category — the model or data is not displayed correctly.

MatMT committed:
fix: Create Default Values

Javithor360 commented:
Thanks for the demo showcase. The new changes look really good and I see that most of the bugs were already covered here, I can perceive a big upgrade from the last iteration. But I have some suggestions that I'll be sharing below. Nice job 👏

Javithor360 commented on OrganiTask/Util/ColorUtil.cs:
```diff
@@ -37,4 +37,14 @@ public static System.Drawing.Color ParseColor(string colorString)
         }
     }
+
+    public static class ColorHelper
```
I don't quite understand why you created another static class inside the `ColorUtil` one just for one method. It would be better if you merge this `IsDarkColor` method into the already existing `ColorUtil` static class.

Javithor360 commented on OrganiTask/Forms/Controls/TaskCardPanel.cs:
```diff
@@ -16,7 +17,7 @@ public class TaskCardPanel : Panel
         {
             this.Width = 240;
-            this.Height = 80;
+            this.Height = 140;
```
I liked the idea of expanding the default task card size as I suggested for longer descriptions. But, now for short descriptions this feels odd. I suggest two possible fixes:
1. Align short descriptions centered horizontally and vertically
2. Adapt the card height dinamically depending on the size of the given description (preferred)

MatMT replied:
It's pending, I'll work on it later because it's complicated for me to learn how to modify dynamic form heights.

Javithor360 commented on OrganiTask/Forms/DashboardSettings.cs:
```diff
@@ -54,6 +46,10 @@ private void LoadDashboardInfo()
             string username = controller.GetUsernameFromDashboardOwnerId(dvm.UserId);
 
             // Cargamos la información del tablero en los controles
+            lblHeader.MaximumSize = new Size(495, 0);
```
Somehow here, in `DashboardSettings`, I see that dashboard descriptions height is adapted based on the description size. This same logic should be applied for each `TaskDetails` form as well but for the moment this doesn't occur and the text doesn't display as expected.

![image](https://github.com/user-attachments/assets/58a31c4b-ea7e-4241-b742-fe6bd8598a11)

MatMT replied:
In the TaskDetails form, I decided to replace the Label with a TextBox and add a default message when there is no description yet.

Javithor360 commented on OrganiTask/Forms/Controls/KanbanColumnPanel.cs:
```diff
@@ -53,7 +68,8 @@ public KanbanColumnPanel(Tag tag)
                 Dock = DockStyle.Fill,
-                AutoEllipsis = true
+                AutoEllipsis = true,
+                ForeColor = ColorHelper.IsDarkColor(baseColor) ? Color.White : Color.Black
```
I loved the idea of detecting dark colors to adapt the font color. Good implementation, this also can be applied for those tags that display in the `TaskDetails` form and the `CategorySettings` form as well since all the tags rendered here have their own background color that may require to be adapted.

![image](https://github.com/user-attachments/assets/20c36f06-3b74-47cc-861c-264876aa917f)

Javithor360 commented on OrganiTask/Controllers/DashboardController.cs:
```diff
@@ -557,5 +559,30 @@ public void DeleteCategory(int categoryId)
             }
         }
+
+        // Añade este método a tu clase DashboardController
```
remove this

Javithor360 commented on OrganiTask/Forms/DashboardsManagement.cs:
```diff
@@ -152,13 +132,40 @@ private void createDefaultValues(int dashboardId)
                         Name = item.ToString(),
                         CategoryId = currentCategory.Id
```
Maybe here you can set default tag colors **just** for the default category tags

Javithor360 replied:
@MatMT consider this for the next iteration

Javithor360 commented on OrganiTask/Forms/DashboardsManagement.cs:
```diff
@@ -22,22 +22,10 @@ public partial class DashboardsManagement : Form
-            }
-        }
-
         private void btnSaveDashboard_Click(object sender, EventArgs e)
```
You missed the cancel button click listener here

Javithor360 commented on OrganiTask/Controllers/DashboardController.cs:
```diff
@@ -557,5 +559,30 @@ public void DeleteCategory(int categoryId)
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Error al verificar si existe la categoría: {ex.Message}");
```
Console class is not used at all in the project. Maybe show MessageBox Dialog instead or cancel the operation.

Javithor360 commented on OrganiTask/Forms/DashboardsManagement.cs:
```diff
@@ -152,13 +132,40 @@ private void createDefaultValues(int dashboardId)
             }
         }
+
+        private void btnAddTag_Click(object sender, EventArgs e)
```
Add comments about the functionality here

Javithor360 commented on OrganiTask/Forms/DashboardsManagement.cs:
```diff
@@ -152,13 +132,40 @@ private void createDefaultValues(int dashboardId)
+            txtTagName.Clear();
+        }
+
+        private void listBoxTags_MouseDoubleClick(object sender, MouseEventArgs e)
```
Add comments about the functionality here

Javithor360 commented on OrganiTask/Forms/KanbanDashboard.cs:
```diff
@@ -328,6 +402,8 @@ private void Card_MouseUp(object sender, MouseEventArgs e)
 
         private void EventRefreshDashboard(object sender, EventArgs e)
         {
+            Console.WriteLine("La información cambio");
```
Remove this

Javithor360 reviewed (changes requested):
As I said earlier, these changes look promising. I liked the `DashboardManager` solution for the dashboards instances, also your code readability has improved and you've kept similar logic with the existing one.

Just check the entire review to proceed. Let me know if you have doubts or want some help!

Nicely done 👏

MatMT committed:
fixed: Changes from the Pull Request have been applied

pending: The dinamyc height feature will be further develop later to apply to other elements as wekk ('Show All Tag's feature')

Javithor360 reviewed (approved):
Well done! 🚀
